assets/js/theme/common/carousel.js

```diff
@@ -2,20 +2,44 @@ import 'slick-carousel';
 
 export default function () {
     const $carousel = $('[data-slick]');
-    if ($carousel.length) {
-        const multipleSlides = $carousel[0].childElementCount > 1;
-        $carousel.slick({ dots: multipleSlides });
-    }
+
+    if ($carousel.length === 0) return;
+
+    const multipleSlides = $carousel[0].childElementCount > 1;
+    $carousel.slick({ dots: multipleSlides });
+
+    const $slidesNodes = $('.heroCarousel-slide');
+
+    $slidesNodes.each((index, element) => {
+        const $element = $(element);
+        const isContentBlock = !!$element.find('.heroCarousel-content').length;
+
+        if (isContentBlock) return true;
+
+        const $image = $element.find('.heroCarousel-image-wrapper img');
+        $('<img/>') // Make in memory copy of image to avoid css issues
+            .attr('src', $($image).attr('src'))
+            .load(function getImageSizes() {
+                const imageRealWidth = this.width; // Note: $(this).width() will not
+                const imageRealHeight = this.height; // work for in memory images.
+
+                const imageAspectRatio = imageRealHeight / imageRealWidth;
+
+                const isImageSquare = imageAspectRatio > 0.8 && imageAspectRatio <= 1.2;
+                const isImageVertical = imageAspectRatio > 1.2;
+
+                const slideClass = isImageSquare && 'is-square-image-type'
+                    || isImageVertical && 'is-vertical-image-type'
+                    || '';
+
+                $element.addClass(slideClass);
+            });
```

Comment:
🍹 small proposal
```suggestion
            // You don't need to specify name of call back function in this case.
            .load(function () {
                const imageAspectRatio = this.height / this.width;
                
                switch (true) {
                    case imageAspectRatio > 0.8 && imageAspectRatio <= 1.2:
                        $element.addClass('is-square-image-type');
                        break;
                    case imageAspectRatio > 1.2:
                        $element.addClass('is-vertical-image-type');
                        break;
                }
            });
```

Reply:
My small proposal for small proposal:
```suggestion
            .load(function () {
                const imageAspectRatio = this.height / this.width;
                
                $element.addClass(() => {
                    switch(true){
                        case imageAspectRatio > 0.8 && imageAspectRatio <= 1.2:
                            return 'is-square-image-type';
                        case imageAspectRatio > 1.2:
                            return 'is-vertical-image-type'
                        default: 
                            return ''
                    }
                })
            });
```

Reply:
Good idea, but I think that adding of the empty row is a little bit redundant.

Reply:
addClass method interface expects string as classname type, so in my opinion it's good to follow it

Reply:
You right, but why we need to call `addClass` method for the case when we don't need to add a class name? 🤔 

I marked this comment as 🍹 (optional), so I do not have a strict position on this piece of code. Anyway, if this code will be optimized I will be happy 😄

Reply:
Thank you for a great review!)

Reply:
Thanks for your proposals!